src/universal/mutations/RemoveOrgUserMutation.js

```diff
@@ -0,0 +1,182 @@
+
+  const {pathname} = location;
+  for (let ii = 0; ii < teamIds.length; ii++) {
+    const teamId = teamIds[ii];
+    const onExTeamRoute = Boolean(matchPath(pathname, {
+      path: `(/team/${teamId}|/meeting/${teamId})`
+    }));
+    if (onExTeamRoute) {
+      history.push('/me');
+      return;
+    }
+  }
+};
+
+export const removeOrgUserOrganizationUpdater = (payload, store, viewerId) => {
+  const removedUserId = getInProxy(payload, 'user', 'id');
+  const orgId = getInProxy(payload, 'organization', 'id');
+  if (removedUserId === viewerId) {
+    handleRemoveOrganization(orgId, store, viewerId);
+  } else {
+    handleRemoveOrgMembers(orgId, removedUserId, store);
+  }
+};
+
+export const removeOrgUserNotificationUpdater = (payload, store, viewerId, options) => {
+  const removedTeamNotifications = payload.getLinkedRecords('removedTeamNotifications');
+  const teamNotificationIds = getInProxy(removedTeamNotifications, 'id');
+  handleRemoveNotifications(teamNotificationIds, store, viewerId);
+
+  const removedOrgNotifications = payload.getLinkedRecords('removedOrgNotifications');
+  const orgNotificationIds = getInProxy(removedOrgNotifications, 'id');
+  handleRemoveNotifications(orgNotificationIds, store, viewerId);
+
+  const kickOutNotifications = payload.getLinkedRecords('kickOutNotifications');
+  handleAddNotifications(kickOutNotifications, payload, viewerId);
+
+  popKickedOutToast(payload, options);
+};
+
+export const removeOrgUserTeamUpdater = (payload, store, viewerId) => {
+  const removedUserId = getInProxy(payload, 'user', 'id');
+  if (removedUserId === viewerId) {
+    const teams = payload.getLinkedRecords('teams');
+    const teamIds = getInProxy(teams, 'id');
+    handleRemoveTeams(teamIds, store, viewerId);
+  }
+};
+
+export const removeOrgUserTeamMemberUpdater = (payload, store, viewerId) => {
+  const removedUserId = getInProxy(payload, 'user', 'id');
+  if (removedUserId === viewerId) {
+    const teamMembers = payload.getLinkedRecords('teamMembers');
+    const teamMemberIds = getInProxy(teamMembers, 'id');
+    handleRemoveTeamMembers(teamMemberIds, store);
+  }
+};
+
+export const removeOrgUserProjectUpdater = (payload, store, viewerId) => {
+  const removedUserId = getInProxy(payload, 'user', 'id');
+  if (removedUserId === viewerId) {
+    const projects = payload.getLinkedRecords('updatedProjects');
+    const projectIds = getInProxy(projects, 'id');
+    handleRemoveProjects(projectIds, store, viewerId);
+  }
+};
+
+const RemoveOrgUserMutation = (environment, orgId, userId, onError, onCompleted) => {
+  const {viewerId} = environment;
+  return commitMutation(environment, {
+    mutation,
+    variables: {orgId, userId},
+    updater: (store) => {
+      const payload = store.getRootField('removeOrgUser');
+      removeOrgUserOrganizationUpdater(payload, store, viewerId);
+    },
+    optimisticUpdater: (store) => {
+      handleRemoveOrganization(orgId, store, viewerId);
+    },
+    onCompleted,
```

Comment:
I'd like to see the "popping the toast" moved from within the `updater` function to the `onCompleted` callback; seems to me that updaters should be focused on updating the store, and all other effects should be created as a result, not as part of the "updating".

Reply:
oh i hear ya! doing so is a HUGE chore. basically, we need to move it to both
subscription.onNext and `mutation.onCompleted`. BUT, here's the kicker with that pattern-- what if we need data from relay to know if we need to pop the toast? For example, store.foo == 1. payload.foo == 1. Pop the toast if payload.foo != store.foo. In english, we don't wanna tell a user that a new team member joined if they already know that. that's annoying (and SO many websites get this wrong...)